server/users.ts

```diff
@@ -521,74 +536,53 @@ class User extends Chat.MessageContext {
 		this.lastMessage = ``;
 		this.lastMessageTime = 0;
 		this.lastReportTime = 0;
-		/**@type {string} */
 		this.s1 = '';
-		/**@type {string} */
 		this.s2 = '';
-		/**@type {string} */
 		this.s3 = '';
 
 		this.blockChallengesNotified = false;
 		this.blockPMsNotified = false;
-		/** @type {boolean} */
 		this.punishmentNotified = false;
-		/** @type {boolean} */
 		this.lockNotified = false;
-		/**@type {string} */
 		this.autoconfirmed = '';
+		this.trusted = '';
 		// Used in punishments
-		/** @type {string} */
 		this.trackRename = '';
-		/** @type {StatusType} */
 		this.statusType = 'online';
-		/** @type {string} */
 		this.userMessage = '';
-		/** @type {number} */
 		this.lastWarnedAt = 0;
 
 		// initialize
 		Users.add(this);
 	}
 
-	/**
-	 * @param {string | BasicRoom?} roomid
-	 * @param {string} data
-	 */
-	sendTo(roomid, data) {
-		// @ts-ignore
-		if (roomid && roomid.id) roomid = roomid.id;
+	sendTo(roomid: string | BasicRoom | null, data: string) {
+		if (roomid && (roomid as BasicRoom).id) roomid = (roomid as BasicRoom).id;
```

Comment:
Ideally you don't do `roomid = roomid` ever.

```suggestion
		if (roomid && typeof roomid !== 'string') roomid = (roomid as BasicRoom).id;
```